Core/GDCore/Events/Expression.h

```diff
@@ -24,32 +29,54 @@ class GD_CORE_API Expression {
   /**
    * \brief Construct an empty expression
    */
-  Expression(){};
+  Expression();
 
   /**
    * \brief Construct an expression from a string
    */
-  Expression(gd::String plainString_) : plainString(plainString_){};
+  Expression(gd::String plainString_);
 
   /**
    * \brief Construct an expression from a const char *
    */
-  Expression(const char* plainString_) : plainString(plainString_){};
+  Expression(const char* plainString_);
+
+  /**
+   * \brief Copy construct an expression.
+   */
+  Expression(const Expression& copy);
+
+  /**
+   * \brief Expression affectation overriding.
+   */
+  Expression& operator=(const Expression& expression);
 
   /**
    * \brief Get the plain string representing the expression
    */
   inline const gd::String& GetPlainString() const { return plainString; };
 
+  /**
+   * @brief Get the expression node.
+   *
+   * @param type the expression type
+   * @param parser the parser to use if this expression never has been parsed
+   * before.
+   * @return std::unique_ptr<gd::ExpressionNode>
+   */
+  std::unique_ptr<gd::ExpressionNode> GetRootNode(
+      const gd::String& type, gd::ExpressionParser2& parser) const;

```

Comment:
The type should probably be passed by the constructor, but it seems like a huge refactor.